src/Codec/Xlsx/Types/Common.hs

```diff
@@ -89,9 +94,27 @@ import Data.Profunctor(dimap)
 import Control.Lens(makePrisms)
 #endif
 
+newtype RowIndex = RowIndex {unRowIndex :: Int}
+  deriving (Eq, Ord, Show, Read, Generic, Num, Real, Enum, Integral)
+newtype ColumnIndex = ColumnIndex {unColumnIndex :: Int}
+  deriving (Eq, Ord, Show, Read, Generic, Num, Real, Enum, Integral)
+instance NFData RowIndex
+instance NFData ColumnIndex
+
+instance ToAttrVal RowIndex where
+  toAttrVal = toAttrVal . unRowIndex
+
+ -- | DEPRECATED: this function will be removed in an upcoming release, use
```

Comment:
What about using pragmas instead? https://downloads.haskell.org/~ghc/latest/docs/users_guide/exts/pragmas.html?#warning-and-deprecated-pragmas - I think they are automatically added to haddocks as well